source/extensions/tracers/xray/localized_sampling.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "envoy/common/time.h"
+#include "envoy/runtime/runtime.h"
+
+#include "common/common/logger.h"
+
+#include "extensions/tracers/xray/reservoir.h"
+#include "extensions/tracers/xray/sampling_strategy.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+class LocalizedSamplingRule {
+public:
+  LocalizedSamplingRule(unsigned fixed_target, unsigned rate)
+      : fixed_target_(fixed_target), rate_(rate), reservoir_(fixed_target_) {}
+
+  bool appliesTo(const SamplingRequest& request) const;
+  static LocalizedSamplingRule createDefault();
+
+  void setHost(absl::string_view host) { host_ = static_cast<std::string>(host); }
```

Comment:
I've never static-casted from string_view to string. Is this done much in the codebase? It seems wrong to me.

Reply:
The one you're used to seeing are functional-style cast. They're equivalent. I tend to dislike functional-style cast a bit because I think they're easily confused with constructors. But, in this case, I'm going to change it to avoid confusion.